src/viur/core/module.py

```diff
@@ -186,23 +186,30 @@ def parse_value_by_annotation(annotation: type, name: str, value: str | list | t
 
                 raise errors.NotAcceptable(f"Missing required parameter {param_name!r}")
 
-        # Extend args to any varargs
-        parsed_args += varargs
-
-        # extend all kwargs if varkwargs is present
-        if varkwargs:
-            for name, value in kwargs.items():
-                if name not in self.signature.parameters:
-                    parsed_kwargs[name] = value
-            varkwargs = bool(kwargs)
-        # always take "skey" when configured into kwargs, if varkwargs is unset
-        elif self.skey and self.skey["name"] in kwargs:
-            parsed_kwargs[self.skey["name"]] = kwargs[self.skey["name"]]
-            varkwargs = True
-
-        args = tuple(parsed_args)
-        kwargs = parsed_kwargs
-
+        # Here's a short clarification on the variables used here:
+        #
+        # - parsed_args     = tuple of (the type-parsed) arguments that have been assigned based on the signature
+        # - parsed_kwargs   = dict of (the type-parsed) keyword arguments that have been assigned based on the signature
+        # - args            = either parsed_args, or parsed_args + remaining args if the function accepts *args
+        # - kwargs          = either parsed_kwars, or parsed_kwargs | remaining kwargs if the function accepts **kwargs
+        # - varargs         = indicator that the args also contain variable args (*args)
+        # - varkwards       = indicator that variable kwargs (**kwargs) are also contained in the kwargs
+        #
+
+        # Extend args to any varargs, and redefine args
+        args = tuple(parsed_args + varargs)
+
+        # always take "skey"-parameter name, when configured, as parsed_kwargs
+        if self.skey and self.skey["name"] in kwargs:
+            parsed_kwargs[self.skey["name"]] = kwargs.pop(self.skey["name"])
+
+        # When varkwargs are accepted, merge parsed_kwargs and kwargs, otherwise just use parsed_kwargs
+        if varkwargs := varkwargs and bool(kwargs):
+            kwargs = parsed_kwargs | kwargs
```

Comment:
I want to keep the key order of parsed_kwargs by their signature. Therfore it is done this way. It would work also with your change, but I like it more when the stuff from the signature stays first in the dict.